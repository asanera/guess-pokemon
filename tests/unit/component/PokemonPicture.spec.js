import PokemonPicture from "@/components/PokemonPicture"
import { shallowMount } from "@vue/test-utils"

describe('PokemonPicture', () => {

    let wrapper;
    beforeAll( async () => {
        wrapper = shallowMount(PokemonPicture)
        await wrapper.setProps({pokemonId: 100})
        await wrapper.setProps({showPokemon: false})
    })

    test('Match con snapshot', () => {
        expect(wrapper.element).toMatchSnapshot();
    })

    test('Debe mostrar la imagen oculta del pokemon 100', () => {
        wrapper = shallowMount(PokemonPicture)
        expect(wrapper.find('.hidden-pokemon')).not.toBeNull();
    })

    test('Debe mostrar el pokemon si showPokemon esta a true', async () =>{
        await wrapper.setProps({showPokemon: true})
        expect(wrapper.findAll('img')[1]).not.toBeNull();
    })
})