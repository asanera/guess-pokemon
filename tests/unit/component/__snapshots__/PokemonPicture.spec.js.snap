// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonPicture Match con snapshot 1`] = `
<div
  class="pokemon-container"
>
  <img
    alt="pokemon"
    class="hidden-pokemon"
    src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/100.svg"
  />
  <!--v-if-->
</div>
`;
